src/Stratis.Bitcoin/P2P/Peer/NetworkPeerConnection.cs

```diff
@@ -136,6 +138,7 @@ private async Task ReceiveMessagesAsync()
                 {
                     Message message = await this.ReadAndParseMessageAsync(this.peer.Version, this.CancellationSource.Token).ConfigureAwait(false);
 
+                    this.asyncProvider.Signals.Publish(new PeerMessageReceived(this.peer.PeerEndPoint, message));
```

Comment:
The same here, perhaps check if the node is in debug and then only publish?

Reply:
same as before